### PR TITLE
Add alias to LMImageAndGlobalCacheJob

### PR DESCRIPTION
**Pull request by Atticus1806**
This PR adds an alias to the LMImageAndGlobalCacheJob within the AdvancedTreeSearchJob. I was not really able to find a way to add this without adding a job input and excluding it from hash. Feel free to suggest better ways if possible.

### CODE REVIEW

File: recognition/advanced_tree_search.py [hunk 1]
```diff
@@ -148,6 +148,8 @@ def hash(cls, kwargs):
 
 
 class AdvancedTreeSearchJob(rasr.RasrCommand, Job):
+    __sis_hash_exclude__ = {"lmgc_alias": None}
+
```
[the remark this answers is not in the document]
Comment (Atticus1806) [reply recognition/advanced_tree_search.py:152]: Oh I did not see that.

**Review by christophmluscher — COMMENTED** [review]
since you are touching this file it would be nice to update typing and so on :)

File: recognition/advanced_tree_search.py [hunk 2]
```diff
@@ -150,25 +151,48 @@ def hash(cls, kwargs):
...
+        lm_lookahead: bool = True,
+        lookahead_options: Optional[Dict[str, Any]] = None,
+        create_lattice: bool = True,
+        eval_single_best: bool = True,
+        eval_best_in_lattice: bool = True,
+        use_gpu: bool = False,
+        rtf: int = 10,
+        mem: int = 4,
+        cpu: int = 1,
+        lmgc_mem: int = 12,
```
Comment (christophmluscher) [inline recognition/advanced_tree_search.py:167]:
```suggestion
        rtf: float = 10.,
        mem: float = 4.,
        cpu: int = 1,
        lmgc_mem: float = 12.,
```

?

File: recognition/advanced_tree_search.py [hunk 3]
```diff
@@ -246,25 +270,28 @@ def update_search_parameters(cls, search_parameters):
...
+        feature_scorer: rasr.FeatureScorer,
+        search_parameters: Union[None, Dict[str, Any]],
+        lm_lookahead: bool,
+        lookahead_options: Union[None, Dict[str, Any]],
+        create_lattice: bool,
+        eval_single_best: bool,
+        eval_best_in_lattice: bool,
+        mem: int,
+        cpu: int,
+        lmgc_mem: int,
```
Comment (christophmluscher) [inline recognition/advanced_tree_search.py:284]:
```suggestion
        mem: float,
        cpu: int,
        lmgc_mem: float,
```

**Review by christophmluscher — APPROVED** [review]
thx! :)

IIRC some params can be `float` instead of `int`. otherwise fine.

**Review by christophmluscher — CHANGES_REQUESTED** [review]
I ll add a comment in a sec

**Review by christophmluscher — APPROVED** [review]
just as a suggestion

**Pull request merged**